assets/php/virion.php

```diff
@@ -193,3 +198,36 @@ function change_dna(string $chromosome, string $antigen, string $antibody, $mode
 
     return $ret;
 }
+
+final class TempDir {
+
+    /** @var string */
+    public $path;
+
+    public function __construct() {
+        $this->path = tempnam(sys_get_temp_dir(), "virion-inject");
+        unlink($this->path);
+        $this->path .= DIRECTORY_SEPARATOR;
+        mkdir($this->path, 0666);
+    }
```

Comment:
Please make this error-safe. Consider race conditions of path being used by another process between unlink() and mkdir().

Reply:
Now you have the chance of two virions building in the same directory and mixing up.

Reply:
I can't think of a way to do this without either throwing an exception (`if(!mkdir()) throw $e`) or a possibly long-lasting `while(!mkdir()) sleep()` loop. I thought of going with `if(!is_dir()) mkdir()` because the only other mkdir call in poggit/poggit goes with that approach, any ideas?

Reply:
Choose another tempnam if the directory creation failed. Only run the subsequent code if you are the one who created this directory.

Reply:
@Muqsit you had a chance to do this ?

Reply:
I am not confident that the logic is still correct... If `$this->path` doesn't already exist, something might have gone wrong, and you should choose another tempnam. If `$this->path` is already a directory, this also means something is wrong, and you should also choose another tempnam.

Reply:
While I do understand the issue, I unfortunately don't have much knowledge in this field and would prefer someone else take over this task as deleting the wrong directory could lead to unperceivable results, possibly even out of the realms of this program.

Reply:
what about sticking `getmypid()` on the end of the folder name?